### PR TITLE
Adopting f-strings in Scripts/ dir via flynt

### PR DESCRIPTION
**Pull request by chris-rands**
See issue #2510. Thoughts on readability here?

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
I like the readability - commented on #2005 

As to the practicalities, might want to fix the concatenation issue and re-run black on this file:

```
$ flake8
./Scripts/Restriction/ranacompiler.py:982:31: BLK100 Black would make changes.
./Scripts/Restriction/ranacompiler.py:983:29: ISC003 explicitly concatenated string should be implicitly concatenated
```

We're getting closer to fully black on ``Bio/`` and this is likely to happen elsewhere. I just run flynt+black on another project already using black, and overall it shortened by code 👍

File: Scripts/Restriction/ranacompiler.py [hunk 1]
```diff
@@ -977,10 +977,10 @@ def information_mixer(self, file1, file2, file3):
                 except OverhangError:  # overhang error
                     n = name  # do not include the enzyme
                     if not bl[2]:
-                        print("Anyway, %s is not commercially available.\n" % n)
+                        print(f"Anyway, {n} is not commercially available.\n")
                     else:
                         print(
-                            "Unfortunately, %s is commercially " % n + " available.\n"
```
Comment (peterjc) [inline Scripts/Restriction/ranacompiler.py]: I can see why after flynt we get a flake8 error - might be worth fixing the original now anyway on a tiny pull request?

Comment (peterjc) [reply Scripts/Restriction/ranacompiler.py]: Fixed in #2572

**Comment by peterjc** [comment]
My work on ``update_ncbi_codon_table.py `` in #2506 make this hard to rebase - probably easier to re-run flynt on the Scripts/ folder - I don't think there would be any objection from the point of view of outstanding pull requests (which is a risk on the Bio/ and Tests/ folders).

**Comment by peterjc** [comment]
Of topic, but the commit removing the ``\r`` FASTA code is interesting, _probably_ safe in Python 3 to assume the text mode conversion will have got rid of this.

**Comment by chris-rands** [comment]
> Of topic, but the commit removing the `\r` FASTA code is interesting, _probably_ safe in Python 3 to assume the text mode conversion will have got rid of this.

Yes, I did not mean to include that here, but my thinking exactly. I also thought that these `.replace()` / `assert` calls *might* be expensive since they are O(n) operations (checking all the characters), but when I actually did a quick benchmark using my code from https://github.com/biopython/biopython/pull/1808#issuecomment-433043873 it didn't make any noticeable difference to the speed of `SeqIO.parse()`

File: Scripts/xbbtools/nextorf.py [hunk 2]
```diff
@@ -220,7 +214,7 @@ def Output(self, CDS):
                     n - start + 1,
                 )
             if self.options["gc"]:
-                head = "%s:%s" % (head, self.Gc2(subs))
+                head = f"{head}:{self.Gc2(subs)}"
```
Comment (peterjc) [inline Scripts/xbbtools/nextorf.py]: It is this kind of f-string which I find most jarring - single variable substitutions I'm absolutely happy with, actual code far less so. I think my editor needs a syntax colouring update which would help.

**Comment by chris-rands** [comment]
Any objections to merging this before the next Biopython release?

**Comment by peterjc** [comment]
I think the reaction on #2510 was generally favourable - and ``Scripts/`` isn't updated so often that the code churn / git blame downside isn't significant.

If there are no objections I suggest later this week you rebase this and/or recreate it (probably easier), and we merge it.

**Comment by peterjc** [comment]
Do you want to update this branch Chris?

A git rebase or merge ought to work - but I suspect that given this was an automatically generated set of changes it might be easiest to delete your local ``flynt_test_Scripts`` branch, update to master, make a new ``flynt_test_Scripts`` branch, apply flynt, and force-push it to GitHub to update the pull request.

Up to you - just ask if you want guidance. Thanks.

**Comment by chris-rands** [comment]
I don't see any conflicts? I did a merge 6 days ago https://github.com/biopython/biopython/pull/2511/commits/f4677aa2678179f3661c7212ca8d8542f3f22d12, did i miss something?

**Comment by peterjc** [comment]
Ah "This branch cannot be rebased due to conflicts" but that doesn't mean I can't just squash-merge it. That OK with you?

**Comment by chris-rands** [comment]
Sorry Peter, I'm confused, where do you read that? I see "This branch has no conflicts with the base branch"

**Comment by peterjc** [comment]
Text on the merge buttons, which you wouldn't see.

You evidently don't see the same information, which does confuse discussions:

Screenshot 2020-02-03 12 29 45

Anyway, I will merge this now - thank you!

**Pull request merged**